control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1429,6 +1438,18 @@ func (r *HostedControlPlaneReconciler) reconcilePKI(ctx context.Context, hcp *hy
 	return nil
 }
 
+func (r *HostedControlPlaneReconciler) reconcileCSIDriver(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage, createOrUpdate upsert.CreateOrUpdateFN) error {
+	switch hcp.Spec.Platform.Type {
+	case hyperv1.KubevirtPlatform:
+		err := kubevirtcsi.ReconcileInfra(r.Client, hcp, ctx, createOrUpdate, releaseImage.ComponentImages())
```

Comment:
are other platforms expected to reconcile here too or should the storage operator do that? 
Does this need to be clarified here https://github.com/openshift/enhancements/blob/master/enhancements/storage/storage-hypershift.md?

How does this PR interacts with https://github.com/openshift/hypershift/pull/1698?

Reply:
> are other platforms expected to reconcile here too or should the storage operator do that? Does this need to be clarified here

most likely other drivers should be laid down by the cso and not the hcp or hcco. KubeVirt is a little different at the moment though.

> How does this PR interacts with https://github.com/openshift/hypershift/pull/1698?

We started planning kubevirt csi driver integration before the CSO started their reference implementation with aws. Essentially both our kubevirt csi integration effort and the CSO's effort occurred in parallel. We understood this was happening and plan to (attempt to) converge with the CSO in 4.13 once they have an established pattern for deploying into the HCP that we can follow.

Converging with the CSO might be slightly challenging though, which is something we'll be investigating soon. Due to the nested nature of the driver (requiring api access to both mgmt and guest clusters) kubevirt csi has some unique requirements that other csi drivers (like aws) do not have. 

For now, we'd like to be able to begin exercising the KubeVirt csi driver using the integration method we have here. A decision about how to handle this long term will be addressed before KubeVirt goes GA (4.13+)